### PR TITLE
Add unique id to Person in NetConnect

### LINKED ISSUE #45: Add unique ID for each person

### PR DESCRIPTION
**Pull request by shaunlxw**
Fixes #45.

Changes include:
* Adding Id class and integration into Person
* Update UI to display id
* Integrate id into storage
* Add JUnit tests for id
* Check for duplicate id when reading from storage
* Add support messages, prefixes, methods and classes needed for id operations
* Update delete and edit command to use id

**Note to developers**:
Validation checks on existing storage file (/data/netconnect.json) will fail, as it does not have an id field yet. This will cause the file to be emptied. Delete the file completely and re-run NetConnect again to populate the app with sample data.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -96,14 +97,15 @@ public CommandResult execute(Model model) throws CommandException {
     private static Person createEditedPerson(Person personToEdit, EditPersonDescriptor editPersonDescriptor) {
         assert personToEdit != null;
 
+        Id oldId = personToEdit.getId();
```
Comment (tanjieling0) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: perhaps a better variable name could be idToEdit? I was trying to find the new id as i thought the id was being updated

File: src/main/java/seedu/address/model/person/Id.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's unique id in the address book, smallest id is 1.
+ * Guarantees: immutable; is valid as declared in {@link #isValidId(int)}
+ */
+public class Id {
+
+    public static final String MESSAGE_CONSTRAINTS = "Ids should only be any integer value larger than 0";
```
Comment (tanjieling0) [inline src/main/java/seedu/address/model/person/Id.java:11]: maybe in the string Id can be written as IDs

File: src/main/java/seedu/address/storage/JsonSerializableNetConnect.java [hunk 3]
```diff
@@ -20,6 +20,7 @@
 class JsonSerializableNetConnect {
 
     public static final String MESSAGE_DUPLICATE_PERSON = "Persons list contains duplicate person(s).";
+    public static final String MESSAGE_DUPLICATE_ID = "Persons list contains duplicate id(s).";
```
Comment (tanjieling0) [inline src/main/java/seedu/address/storage/JsonSerializableNetConnect.java:23]: Perhaps can standardise to write IDs as ID or id or Id in strings

**Review by tanjieling0 — COMMENTED** [review]
LGTM except some small cosmetic issues (that do not affect the functionalities) that you may choose to ignore. Solid code and looks very neat and logically coded. You are so cool

**Review by tanjieling0 — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/DeleteCommandParser.java
Comment (nuyer) [inline src/main/java/seedu/address/logic/parser/DeleteCommandParser.java:1]: I'm not very sure if this belong to your part. We can delete contact by both id and name but the implementation is by ID only.

**Review by nuyer — APPROVED** [review]
LGTM👍🏻

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 1 above)
Comment (shaunlxw) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: Hm will change to `idOfPerson` to reflect it better

File: src/main/java/seedu/address/storage/JsonSerializableNetConnect.java (hunk 3 above)
Comment (shaunlxw) [reply src/main/java/seedu/address/storage/JsonSerializableNetConnect.java:23]: I think I standardised to `id` for messages! Some places use `ID` as it is used in parameters, similar to `n/NAME` so `i/ID`. Hopefully it doesn't differ anywhere else hehe

File: src/main/java/seedu/address/model/person/Id.java (hunk 2 above)
Comment (shaunlxw) [reply src/main/java/seedu/address/model/person/Id.java:11]: Standardised as `id`.

File: src/main/java/seedu/address/logic/parser/DeleteCommandParser.java
Comment (shaunlxw) [reply src/main/java/seedu/address/logic/parser/DeleteCommandParser.java:1]: Yeps, that part of the implementation is still missing. Not sure if it is under me 😆 But I'm keeping this PR strictly for ID changes only. Will open another PR for updating the command format for `delete`.

**Pull request merged**